python/tvm/contrib/pipeline_executor.py

```diff
@@ -86,8 +88,43 @@ class PipelineModule(object):
         Common interface for pipeline executor factory modules.
     """
 
-    def __init__(self, module):
-        self.module = module.module
+    def __init__(self, module=None):
+        self.module = module.module if module else None
+        self._get_num_outputs = None
+        # Get the packed functions from the pipeline executor.
+        self.load_functions()
+
+    def import_from_library(self, config_file_name):
+        """Import files to create pipeline executor.
+
+        Parameters
+        ----------
+        config_file_name : str
+            The configuration file path, the configuration file contains the
+            disk path of the parameter file, library file and JSON file。
+        """
+        # Create a empty PipelineExecutorFactoryModule.
+        pipeline_factory = PipelineExecutorFactoryModule()
+        # Load the configuration file to initialize a PipelineExecutorFactoryModule.
+        pipeline_factory.import_from_library(config_file_name)
+        self.module = pipeline_factory.module
+        # Get packed functions from the pipeline executor.
+        self.load_functions()
+
+    def load_functions(self):
+        # Get functions from the pipeline executor.
+        self._get_num_outputs = self.module["get_num_outputs"] if self.module else None
+
+    def get_num_outputs(self):
+        """Get the number of outputs.
+        Returns
+        -------
+        count : int
+            The number of outputs.
+        """
+        if not self._get_num_outputs:
+            raise RuntimeError(f"The pipeline executor has not been initialized.")
+        return self._get_num_outputs()
```

Comment:
_get_num_outputs is packed function returned by self.module["get_num_outputs"], the reason here to check if self._get_num_outputs is none is because when create PipelineModule, the 'module' can be null, then only after the 'load_library' function get called then _get_num_outputs can be valid.

used @property and changed function name into num_outputs.

Reply:
I think this is the key that makes the implementation non-trivial. Can we disallow module to be None when creating a PipelineModule? You should make the initializer static, so that `load_functions` can be removed, such as:

```python
class PipelineModule(object):
  def __init__(self, module):
    self.module = module.module
    self._get_num_outputs = self.module["get_num_outputs"]

  @staticmethod
  def load_library(config_file_name):
    pipeline_factory = PipelineExecutorFactoryModule()
    pipeline_factory.load_library(config_file_name)
    return PipelineModule(pipeline_factory)
```

And the usage:

```python
pipeline_module = PipelineModule(module) # Initialize using a module
pipeline_module = PipelineModule.load_library(config_file) # Initialize using the configure file
```